api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -193,4 +194,21 @@ message RateLimitResponse {
   // - :ref:`envoy.filters.network.ratelimit <config_network_filters_ratelimit_dynamic_metadata>` for network filter.
   // - :ref:`envoy.filters.thrift.rate_limit <config_thrift_filters_rate_limit_dynamic_metadata>` for Thrift filter.
   google.protobuf.Struct dynamic_metadata = 6;
+
+  // Quota is available for a request if its whole descriptor set has cached quota available.
```

Comment:
Question: Is this just a wire optimization? It seems easier to reason about just sending the same quota for each descriptor in the response? I'm mainly trying to understand if we need this extra layer of indirection.

Reply:
This is not an optimization. It depends on the implementation of the RL server. RL can hand out quota for the entire descriptor set (which is how Google's internal RL server works) or for individual descriptors. Envoy's behavior will be different in these cases as described in the comments.

If there is quota associated with the entire descriptor set it is used to admin request. Otherwise per descriptor quota is checked. Otherwise request to RL server is made.

Reply:
I guess I don't understand the difference. Can you add more comments/commentary? Is the idea that the set is unique and composed of all descriptors? So it's basically a lookup using a key of (Desc1 && Desc2 && DecN...)?

Reply:
I don't have any more questions. This makes sense to me, but I think more of this comment thread needs to be documented in the API docs so please make it more clear there what is going on?

/wait

Reply:
Sure. Looks like `Quota` wasn't documented in API docs before. 
I am wondering where you think will be a good place for it?  current.rst ? 
I also found docs/root/configuration/other_features/rate_limit.rst. But it seems pretty empty/obsolete.

Reply:
You can document it in this file. What I'm saying is that the docs in this file I don't think are clear enough about what the response wide quota actually means. I would describe that it's a union of all descriptors and that is used for further matches, etc.